api/types/app.go

```diff
@@ -472,7 +472,7 @@ func (a *AppV3) GetDisplayName() string {
 	if a.Spec.IdentityCenter == nil {
 		return ""
 	}
-	return a.GetName()
+	return a.Metadata.Description
```

Comment:
I needed to change the name value of the "app" to be the account ID - for two reasons:

 1. so I can pull it out to build the dummy Account Assignment that we use to query access
 2. to avoid the uniqueness issues we discussed yesterday

The knock-on effect of this is that the friendly name is now pulled from the app description.